tests/Domain/DomainObjectTest.php

```diff
@@ -219,6 +221,28 @@ public function parserDataSet(): array
                 DomainTransferStatus::class,
                 include __DIR__ . '/data/domain_transfer_status.php',
             ],
+            'valid notification (all fields)' => [
+                Notification::class,
+                include __DIR__ . '/data/notification_valid.php',
+            ],
+            'valid notification (only required)' => [
+                Notification::class,
+                include __DIR__ . '/data/notification_valid_only_required.php',
+            ],
+            'invalid notification (id)' => [
+                Notification::class,
+                include __DIR__ . '/data/notification_invalid_id.php',
+                TypeError::class,
+            ],
+            'valid notification poll' => [
+                NotificationPoll::class,
+                include __DIR__ . '/data/notification_poll_valid.php',
+            ],
+            'invalid notification poll (count)' => [
```

Comment:
I would suggest you also add a case for the NotificationPoll data that does not include a notification object.